taskboot/docker.py

```diff
@@ -343,10 +343,37 @@ def build(self, context_dir, dockerfile, tags, build_args=[]):
                     if progress and "current" in progress and "total" in progress:
                         percent = round(100.0 * progress["current"] / progress["total"])
                         out += f" {percent}%"
+                elif "error" in state:
+                    logger.error(f"DinD build: {state['error']}")
+                    return
+                else:
+                    out = repr(state)
                 logger.info(f"DinD build: {out}")
             except (KeyError, json.decoder.JSONDecodeError):
                 logger.info(f"DinD build: {line}")
 
+        # Process the docker build by reading the client stream
+        # Sometimes the docker daemon does not respond, crashing the inner read code
+        # So we retry a few times before giving up
+        max_try = 5
+        for i in range(1, max_try + 1):
+            try:
+                for line in build_output:
+                    _read_line(line)
+            except http.client.IncompleteRead:
+                logger.error(
+                    f"Error while reading Docker daemon output, on try {i}/{max_try}"
+                )
+                if i == max_try:
+                    raise Exception(
+                        "Failed to build the docker image, too many read errors"
+                    )
+                continue
```

Comment:
is it worth having a `sleep` or an exp-backoff ?

Reply:
Good idea, i've added  a simple `time.sleep `